### PR TITLE
[relates #286] Rest metrics test

### PR DESCRIPTION
**Pull request by Stefan365**
rest test prototype for metrics

### CODE REVIEW

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	public void validateThatNumberOfAllMessagesOfIntegrationIsGreaterThanPeriodInMs(String integrationName, Integer nr, Integer ms) throws InterruptedException {
+		Thread.sleep(ms + 1000);
+//		0.  get integration id.
+		String id = this.getIdByIntegrationName(integrationName);
+		Assertions.assertThat(id).isNotNull();
+
+//		1. get metrics:
+		try {
+			metricsEndpoint = new MetricsEndpoint();
+		} catch (GeneralSecurityException e) {
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: Pls initialize both endpoints in contructor. @tplevko and maybe I'd like to swallow that exception in AbstractEndpoint :)

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	private String getIdByIntegrationName(String integrationName) {
+		try {
+			integrationsEndpoint = new IntegrationsEndpoint();
+		} catch (GeneralSecurityException e) {
+			Assertions.fail("Error: " + e);
+		}
+		List<Integration> integrations = integrationsEndpoint.list();
+		String id = null;
+		for (Integration integration: integrations){
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: Nice place for a bit of functional programming 😈

Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: ok

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java (hunk 1 above)
Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: ok

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		Date lastProceded = summary.getLastProcessed().get();
+		log.info("LAST MESSAGE WAS PROCEEDED ON: *{}*", lastProceded);
+	}
+
+
+//	AUXILIARIES
+
+	private String getIdByIntegrationName(String integrationName) {
+		List<Integration> integrations = integrationsEndpoint.list();
+		Integration integr = integrations.stream().filter(integration -> integrationName.equals(integration.getName())).findAny().orElse(null);
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: Imho you could just put `return` here and it's fine to check `notNull` on line 37 as keep this as value-retrieving method only.

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			integrationsEndpoint = new IntegrationsEndpoint();
+			metricsEndpoint = new MetricsEndpoint();
+		} catch (GeneralSecurityException e) {
+			Assertions.fail("Error: " + e);
+		}
+
+
+	}
+	@Then("^validate that number of all messages through integration \"([^\"]*)\" is greater than \"([^\"]*)\", period in ms: \"([^\"]*)\"$")
+	public void validateThatNumberOfAllMessagesOfIntegrationIsGreaterThanPeriodInMs(String integrationName, Integer nr, Integer ms) throws InterruptedException {
```
Comment (dsimansk) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: I suppose you check all the available values that are provided by metrics endpoint, right?

Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: I'm going to (asap when suitable syndedis instance is available)

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java (hunk 3 above)
Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java]: done

**Comment by dsimansk** [comment]
@Stefan365 pls rebase on master after Tomas's PR and we could merge this eventually. Can be improved further in separate PRs.

**Comment by Stefan365** [comment]
done.

**Pull request merged**